### ISSUE TITLE
Trouble with example code for Bayesian Inference

### ISSUE THREAD

**Issue by yakdahl** [sebapersson/PEtab.jl#194]
Hi! I'm Jakob, a chemistry collaborator of Torkel's, and I was trying to see if I could figure out a way to utilize the Bayesian Inference implemented in PEtab for Catalyst (I really like the combination of the packages - I find this very exciting!). Unfortunately, I'm already getting into trouble with the example code in the documentation. The setup works fine, but I get the following MethodError when trying to construct the PEtabLogDensity: 

target = PEtabLogDensity(petab_problem)

no method matching PEtabLogDensity(::PEtabODEProblem{ ......

Closest candidates are:
  PEtabLogDensity(::T, ::Any, ::Any, ::Vector{T2}, ::I) where {T<:PEtab.InferenceInfo, I<:Integer, T2<:AbstractFloat}
   @ PEtab C:\Users\Jakob Dahl\.julia\packages\PEtab\sQu6O\src\Structs.jl:924

Any ideas what's going wrong here? Briefly checked the package and I can't find any function that reads a PEtabLogDensity from a PEtabODEProblem as suggested in the docs but I'm not a very experienced julia user, so it's totally possible that I'm doing something else wrong.

**Comment by sebapersson** [comment]
Hi Jakob, 

As Bayesian inference is an extension for PEtab.jl, I think you have forgotten to load the extension packages needed to construct the `PEtabLogDensity`. The following should work:

```julia
using Bijectors, LogDensityProblems, LogDensityProblemsAD
target = PEtabLogDensity(petab_problem)
```

And this example runs locally for me:

```julia
using ModelingToolkit, OrdinaryDiffEq, PEtab, Plots, Bijectors, LogDensityProblems, 
      LogDensityProblemsAD, DataFrames, Random, Distributions

# Dynamic model
@parameters b1, b2
@variables t x(t)
D = Differential(t)
eqs = [
    D(x) ~ b2*(b1 - x)
]
specie_map = [x => 0]
parameter_map = [b1 => 1.0, b2 => 0.2]
@named sys = ODESystem(eqs)

# Simulate data
Random.seed!(1234)
oprob = ODEProblem(sys, specie_map, (0.0, 2.5), parameter_map)
tsave = collect(range(0.0, 2.5, 101))
dist = Normal(0.0, 0.03)
_sol = solve(oprob, Rodas4(), abstol=1e-12, reltol=1e-12, saveat=tsave, tstops=tsave)
# Normal measurement noise with σ = 0.03
obs = _sol[:x] .+ rand(Normal(0.0, 0.03), length(tsave))
plot(_sol.t, obs, seriestype=:scatter, title = "Observed data")

# Measurement data
measurements = DataFrame(
    obs_id="obs_X",
    time=_sol.t,
    measurement=obs)
# Observable
@parameters sigma
obs_X = PEtabObservable(x, sigma)
observables = Dict("obs_X" => obs_X)

_b1 = PEtabParameter(b1, value=1.0, lb=0.0, ub=5.0, scale=:log10, prior_on_linear_scale=true, prior=Uniform(0.0, 5.0))
_b2 = PEtabParameter(b2, value=0.2, scale=:log10, prior_on_linear_scale=false, prior=Uniform(-6, log10(5.0)))
_sigma = PEtabParameter(sigma, value=0.03, lb=1e-3, ub=1e2, scale=:lin, prior_on_linear_scale=true, prior=Gamma(1.0, 1.0))
parameters_ets = [_b1, _b2, _sigma]

petab_model = PEtabModel(sys, observables, measurements, parameters_ets; state_map=specie_map, verbose=false)
petab_problem = PEtabODEProblem(petab_model; ode_solver=ODESolver(Rodas5(), abstol=1e-6, reltol=1e-6), verbose=false)

target = PEtabLogDensity(petab_problem)
xpetab = petab_problem.θ_nominalT
xprior = to_prior_scale(petab_problem.θ_nominalT, target)
xinference = target.inference_info.bijectors(xprior)

using AdvancedHMC
# δ=0.8 - acceptance rate (default in Stan)
sampler = NUTS(0.8)
Random.seed!(1234)
res = sample(target, sampler,
             2000;
             nadapts = 1000,
             initial_params = xinference,
             drop_warmup=true,
             progress=false)

using MCMCChains
chain_hmc = PEtab.to_chains(res, target)
```

As for the actual inference, unless your model is very unidentifiable and large (more than 15 ODEs), I would recommend to use the `NUTS` sampler as above (we are currently doing some benchmarks, and it performs well :)

Hopefully this fixes your problem?

**Comment by yakdahl** [comment]
Thank you! You were right that it was a package problem - I had loaded all the packages but I'm still on Julia 1.9 and for some reason this downloaded PEtab 2.0.0 as a standard - which didn't have the proper extensions implemented yet, correct? Once I specified the current stable release via Pkg.add(PackageSpec(name="PEtab", version="2.14.2")), the example worked as shown. I'll take a look at connecting the Bayesian inference to Catalyst at some point later this month, hopefully on Julia 1.10 .
